application/templates/cms/edit_measure_version.html

```diff
@@ -211,6 +193,73 @@ <h3 class="govuk-heading-m">Updates and corrections</h3>
         </div>
     </form>
 
+    <div class="govuk-grid-row govuk-!-margin-bottom-5">
+      <div class="govuk-grid-column-two-thirds">
+        <div class="govuk-form-group {% if data_sources_not_added %}govuk-form-group--error{% endif %}">
+          <h2 class="govuk-heading-m" id="data-sources">Data sources</h2>
+          <div class="govuk-!-margin-bottom-5">
+            {% if new %}
+              <p class="govuk-body">Once this page is saved you will be able to add data sources</p>
+            {% else %}
+              <h3 class="govuk-heading-s">Primary data source</h3>
+
+              {% if measure_version.primary_data_source %}
```

Comment:
I wonder if we could move all this to a partial that can be reused for the secondary data source?

Reply:
Good idea; done!